mina.netty/src/main/java/org/kaazing/mina/core/service/AbstractIoAcceptorEx.java

```diff
@@ -65,7 +65,7 @@ public BindFuture bindAsync(final SocketAddress localAddress) {
             public void operationComplete(BindFuture future) {
                 if (future.isBound()) {
                     boolean activate = false;
-                    synchronized (bindLock) {
+                    synchronized (boundAddresses) {
```

Comment:
There are other places where boundAddresses is modified across threads. This is not that simple.